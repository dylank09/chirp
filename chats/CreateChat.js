import React, { useState } from "react";
import { StyleSheet, View, Text } from "react-native";
import { AntDesign } from "@expo/vector-icons";

import { theme } from "../assets/Theme";
import ChirpButton from "../components/ChirpButton";
import ChirpTextBox from "../components/ChirpTextBox";

import firebase from "firebase/app";
import "firebase/firestore";
import "firebase/auth";

const firestore = firebase.firestore();
const auth = firebase.auth();

export default function CreateChat({ onBackPress }) {
  const [name, setName] = useState("");

  function createChat() {
    const chatsRef = firestore.collection("chatGroups");
    chatsRef.doc().set({
      name: name,
      createdAt: firebase.firestore.FieldValue.serverTimestamp(),
      members: [auth.currentUser.email],
      membersUnseen: [],
      lastMessage: "",
      lastMessageTimestamp: null,
    });

    onBackPress();
  }

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <AntDesign
          style={styles.back}
          name="left"
          size={24}
          color="white"
          onPress={onBackPress}
        />
        <Text style={styles.chatName}>Create new chat</Text>
      </View>
      <View style={styles.chatInfo}>
        <ChirpTextBox
          placeholder="Name"
          value={name}
          onChangeText={setName}
        ></ChirpTextBox>
      </View>
      <ChirpButton
        onPress={createChat}
        width="70%"
        text="Create Chat"
      ></ChirpButton>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    width: "100%",
    height: "100%",
    backgroundColor: theme.colors.background,
    alignItems: "center",
    justifyContent: "space-between",
  },
  header: {
    flexDirection: "row",
    width: "100%",
    justifyContent: "space-between",
    marginVertical: 15,
  },
  back: {
    marginLeft: 10,
  },
  chatName: {
    color: theme.colors.text,
    width: "100%",
    textAlign: "center",
    marginRight: 25,
    fontWeight: "bold",
    fontSize: theme.dimensions.standardFontSize,
  },
  chatInfo: {
    height: "60%",
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
});
