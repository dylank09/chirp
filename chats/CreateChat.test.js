import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";

import CreateChat from "./CreateChat";

describe("<CreateChat />", () => {
  const tree = renderer.create(<CreateChat />).toJSON();

  it("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  it("input name of chat box is present", () => {
    const { getByTestId } = render(<CreateChat />);
    const foundTextBox = getByTestId("nameBox");
    expect(foundTextBox).toBeTruthy();
  });

  it("create button is present", () => {
    const { getByTestId } = render(<CreateChat />);
    const foundButton = getByTestId("createChatButton");
    expect(foundButton).toBeTruthy();
  });
});
