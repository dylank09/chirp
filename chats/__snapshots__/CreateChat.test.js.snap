// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateChat /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(76, 76, 71)",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 20,
        "marginTop": 20,
        "width": "90%",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 17,
          "fontWeight": "bold",
        }
      }
    >
      Create Chat
    </Text>
    <Text />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "60%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      blurOnSubmit={true}
      multiline={false}
      onChangeText={[Function]}
      placeholder="Name"
      placeholderTextColor="rgb(158, 158, 158)"
      rejectResponderTermination={true}
      secureTextEntry={false}
      selectTextOnFocus={false}
      selectionColor="rgb(193, 73, 83)"
      style={
        Array [
          Object {
            "borderBottomColor": "rgb(255, 255, 255)",
            "borderBottomWidth": 1,
            "color": "rgb(255, 255, 255)",
            "width": "82%",
          },
          Object {
            "fontSize": 15,
            "marginBottom": 4,
            "marginTop": 7,
            "padding": 5,
          },
        ]
      }
      testID="nameBox"
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "color": "rgb(193, 73, 83)",
          "fontSize": 12,
          "height": 15,
          "margin": 0,
          "paddingLeft": 3,
          "width": "75%",
        }
      }
    >
       
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(193, 73, 83)",
        "borderRadius": 18,
        "height": undefined,
        "margin": 10,
        "opacity": 1,
        "padding": 10,
        "width": "70%",
      }
    }
    testID="createChatButton"
  >
    <Text
      style={
        Object {
          "color": "rgb(255, 255, 255)",
          "fontSize": 15,
        }
      }
    >
      Create Chat
       
    </Text>
  </View>
</View>
`;
